streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -773,14 +776,21 @@ private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
         final GlobalStateStoreProvider globalStateStoreProvider = new GlobalStateStoreProvider(internalTopologyBuilder.globalStateStores());
         queryableStoreProvider = new QueryableStoreProvider(storeProviders, globalStateStoreProvider);
 
-        stateDirCleaner = Executors.newSingleThreadScheduledExecutor(r -> {
+        stateDirCleaner = setupStateDirCleaner();
+
+        maybeWarnAboutCodeInRocksDBConfigSetter(log, config);
+        rocksDBMetricsRecordingService = maybeCreateRocksDBMetricsRecordingService(clientId, config);
+    }
+
+    private ScheduledExecutorService setupStateDirCleaner() {
+        return Executors.newSingleThreadScheduledExecutor(r -> {
             final Thread thread = new Thread(r, clientId + "-CleanupThread");
             thread.setDaemon(true);
+
+            ClientMetrics.addNumAliveCleanupThreadMetric(streamsMetrics, (metricsConfig, now) -> thread.isAlive() ? 1 : 0);
```

Comment:
req: Please add verifications or unit tests to verify the values measured by this metric in `KafkaStreamsTest`.

Reply:
The `KafkaStreamsTest` has mocks on all of its modules including metrics, I'll do this verification in `MetricsIntegrationTest`.